galley/pkg/config/analysis/analyzers/injection/injection.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package injection
+
+import (
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+
+	"istio.io/istio/galley/pkg/config/analysis"
+	"istio.io/istio/galley/pkg/config/analysis/msg"
+	"istio.io/istio/galley/pkg/config/processor/metadata"
+	"istio.io/istio/galley/pkg/config/resource"
+)
+
+// Analyzer checks conditions related to Istio injection
+type Analyzer struct{}
+
+var _ analysis.Analyzer = &Analyzer{}
+
+// Name implements Analyzer
+func (s *Analyzer) Name() string {
+	return "injection.Analyzer"
+}
+
+// We assume that enablement is via an istio-injection=enabled namespace label
+// In theory, there can be alternatives using Mutatingwebhookconfiguration, but they're very uncommon
+// See https://istio.io/docs/ops/troubleshooting/injection/ for more info.
+const injectionLabelName = "istio-injection"
+const injectionLabelEnableValue = "enabled"
+
+const istioProxyName = "istio-proxy"
```

Comment:
The injection status and list of containers added can also be discovered by checking the `sidecar.istio.io/status` annotation in the pod.

https://github.com/istio/api/blob/638cf7f5063efe5619cbf296dd83e2b36e784df9/annotation/annotations.yaml#L66-L71

Reply:
So this looks like this:

```json
{
    "version": "610f2b5742375d30d7f484e296fd022086a4c611b5a6b136bcf0758767fefecc",
    "initContainers": [
        "istio-init"
    ],
    "containers": [
        "istio-proxy"
    ],
    "volumes": [
        "istio-envoy",
        "istio-certs"
    ],
    "imagePullSecrets": null
}
```

Is the simple presence of this annotation sufficient to determine that the proxy is running? Or are you suggesting that we look at this first to get the container name, and then look for that specific container?

We'll need the container image later to check that it's the right version.

Reply:
We need the specific image to check versions. You can use the annotation to find the container name(s). This is sufficient for any pod injected by the webhook. istioctl kube-inject also adds the annotation but the user could modify/remove it before deploying.